torchtune/data/_utils.py

```diff
@@ -42,7 +43,43 @@ def truncate(
     return tokens_truncated
 
 
-def split_text_by_image_tag(content: str, image_tag: str) -> List[Dict[str, str]]:
+def load_image(image_path: str) -> "PIL.Image.Image":
+    """
+    Load an image from a file path or a binary stream.
+
+    Args:
+        image_path (str): The image to load from either remote source of local file.
+
+    Raises:
+        ValueError: If the image cannot be loaded from remote source.
+        ValueError: If the image cannot be opened as a PIL.Image.
+
+    Returns:
+        PIL.Image.Image: The loaded image.
+    """
+    # Hackily import PIL to avoid burdensome import in the main module
+    # TODO: Fix this
+    from PIL import Image
+
+    # If a remote source, try to load to local
+    if image_path.startswith("http"):
+        try:
+            image_path = request.urlopen(image_path)
+        except Exception as e:
+            raise ValueError(f"Failed to load image from {image_path}") from e
+
+    # Open the local image as a PIL image
+    try:
+        image = Image.open(image_path)
+    except Exception as e:
+        raise ValueError(f"Failed to open image as PIL Image from {image_path}") from e
+
+    return image
+
+
+def format_content_with_images(
+    content: str, *, image_tag: str, images: List["PIL.Image.Image"]
+) -> List[Dict[str, str]]:
     """
     Given a raw text string, split by the specified ``image_tag``
     and form into list of dictionaries to be used in the ``Message`` content
```

Comment:
nit: while you're here, could you describe what happens if image tag is not found in the content string? IIRC, it's just a no-op

Reply:
So, if image_tag and images are both none, it's a no-op. But otherwise, it'll error b/c if there's images, it expects an image_tag.